caikit/runtime/service_factory.py

```diff
@@ -350,7 +354,12 @@ def _create_request_message_types(
                 continue
 
             decorator = dataobject(package=package_name)
-            cls_ = type(task.request.name, (object,), attrs)
+            cls_ = _DataObjectBaseMetaClass.__new__(
```

Comment:
Same point here about maybe wanting a cleaner way to dynamically create a new `DataObjectBase` derived class on the fly